app/services/purchase_service.rb

```diff
@@ -0,0 +1,35 @@
+class PurchaseService
+  def self.call(purchase_params)
+    return gateway_error unless gateway_valid?(purchase_params)
+    return cart_error unless cart = CartCreator.call(purchase_params.slice(:cart_id, :user))
+    user = cart.user
+    return make_error(user) unless user.valid?
+    order = OrderCreator.call(cart, user, address_params(purchase_params))
+    order.save
+    return make_error(order) unless order.valid?
+    return order
+  end
```

Comment:
Normalmente num service  nós chamamos o call assim 
```suggestion
class PurchaseService
  def self.call(purchase_params)
    new.call(purchase_params)
  end
  
  def call(purchase_params)
    #execução do código
  end
```

Reply:
fazendo isso tu pode colocar todos os metodos como metodos de instancia e ainda pode criar variaveis de instancia pros teus metodos